### ISSUE TITLE
Support dateCreated and dateCreated attributes when saving through code

### ISSUE THREAD

**Issue by pdaleramirez** [craftcms/cms#2909]
https://github.com/craftcms/cms/blob/fc9365c4cbc7f5c678fe788ba423fc65f736d650/src/elements/Entry.php#L1018
```
        $record->sectionId = $this->sectionId;
        $record->typeId = $this->typeId;
        $record->authorId = $this->authorId;
        $record->postDate = $this->postDate;
        $record->expiryDate = $this->expiryDate;
        $record->save(false);
```
We have an importing feature in our plugin that allows user to set dateCreated and dateUpdated on newly imported entries but the `afterSave` method on Entry class does not pass the dateCreated and dateUpdated values on the record class. This prevents it from saving the dateCreated and dateUpdated values that has been set.

**Comment by brandonkelly** [comment]
Those are meant to be more like audit attributes that our base ActiveRecord class sets automatically (and doesn’t even give the outside code a choice on their values), so this isn’t something we could change without making a pretty significant change to ActiveRecord behavior.

So if you really want to hard-set those values, you’d have to either create the element rows manually, or let the Elements service continue doing the initial work, and then go in and update their values after the fact. (Note that for both `insert` and `update` queries, the default behavior is to automatically set those values as well. But at least there you can pass `false` to the `$includeAuditColumns` argument.)

**Comment by pdaleramirez** [comment]
Alright no worries

**Comment by brandonkelly** [comment]
I ended up having a need for this, so have implemented it on the `3.2` branch.
